drivers/flash/qspi_nor_flash: Remove dev null check

de-nordic opened the pull request:
The PR contains two commits:

 - e33123b2f1 drivers/flash/nrf_qspi_nor: Remove qspi_nor_read_id param flash_id;
 - ef6d36e1ec drivers/flash/qspi_nor_flash: Remove dev null check.
 
Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

de-nordic committed:
drivers/flash/qspi_nor_flash: Remove dev null check

The commit removes dev != NULL check from qspi_nrfx_configure.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

anangl commented on drivers/flash/nrf_qspi_nor.c:
```diff
@@ -1109,7 +1106,9 @@ static int qspi_nor_write_protection_set(const struct device *dev,
+	if (!dev) {
+		return -ENXIO;
+	}
 
```
Unnecessary empty line.

de-nordic replied:
Done

anangl commented on drivers/flash/nrf_qspi_nor.c:
```diff
@@ -1109,7 +1106,9 @@ static int qspi_nor_write_protection_set(const struct device *dev,
 static int qspi_nor_configure(const struct device *dev)
 {
-	const struct qspi_nor_config *params = dev->config;
+	if (!dev) {
+		return -ENXIO;
+	}
```
Other drivers don't perform such checks, so perhaps this should be removed completely?

de-nordic replied:
I was also considering that or CHECKIF. I will remove it.

de-nordic replied:
Done

anangl commented on drivers/flash/nrf_qspi_nor.c:
```diff
@@ -1119,6 +1118,8 @@ static int qspi_nor_configure(const struct device *dev)
 
 	ANOMALY_122_UNINIT(dev);
 
+	const struct qspi_nor_config *params = dev->config;
```
What about moving it to `qspi_nor_read_id()` where it is actually needed?

de-nordic replied:
Done

de-nordic committed:
drivers/flash/nrf_qspi_nor: Remove qspi_nor_read_id param flash_id

The commit removes redundant flash_id argument that the function
can obtain itself from dev.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>
